agent/structs/config_entry_gateways.go

```diff
@@ -980,77 +980,6 @@ func (e *BoundAPIGatewayConfigEntry) GetEnterpriseMeta() *acl.EnterpriseMeta {
 	return &e.EnterpriseMeta
 }
 
-func (e *BoundAPIGatewayConfigEntry) UpdateRouteBinding(refs []ResourceReference, route BoundRoute) (bool, map[ResourceReference]error) {
```

Comment:
This functionality was moved to the `GatewayMeta` struct where it has the necessary context to match listener/route protocols.